### PR TITLE
WebConsole support for big numeric SQL types 

### PR DESCRIPTION
**Pull request by Karakatiza666**
Support in Data Browser, Data Import and Data Generator

Fix https://github.com/feldera/feldera/issues/851
Fix https://github.com/feldera/feldera/issues/1006

### CODE REVIEW

File: web-console/src/lib/components/streaming/import/randomData/index.tsx [hunk 1]
```diff
@@ -43,14 +45,14 @@ export const RngFieldSettings = (props: {
...
       true_pct: '0.5',
-      value: '0',
-      min: '0',
-      max: '100',
-      mu: '100',
-      sigma: '10',
-      lambda: '0.1',
-      alpha: '0.1',
-      beta: '0.1',
+      value: new BigNumber('0'),
```
Comment (gz) [inline web-console/src/lib/components/streaming/import/randomData/index.tsx:48]: It looks like now everything is converted to BigNumber.. which is probably fine but it seems like now we need to protect the types that of all ranges correctly (e.g., if it's 64bit we should not be able to have something larger) I didin't see the code yet where this is enforced

File: web-console/src/lib/compositions/streaming/import/useDefaultRows.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export const getDefaultValue = (columntype: ColumnType) =>
   match(columntype)
     .with({ type: 'BOOLEAN' }, () => false)
     .with({ type: 'TINYINT' }, { type: 'SMALLINT' }, { type: 'INTEGER' }, { type: 'BIGINT' }, () => 0)
-    .with({ type: 'DECIMAL' }, { type: 'FLOAT' }, { type: 'DOUBLE' }, () => 0.0)
+    .with({ type: 'DECIMAL' }, { type: 'NUMERIC' }, { type: 'FLOAT' }, { type: 'DOUBLE' }, () => 0.0)
```
Comment (gz) [inline web-console/src/lib/compositions/streaming/import/useDefaultRows.ts:15]: numeric

File: web-console/src/lib/compositions/streaming/import/useDefaultRows.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const getDefaultValue = (columntype: ColumnType) =>
   match(columntype)
     .with({ type: 'BOOLEAN' }, () => false)
     .with({ type: 'TINYINT' }, { type: 'SMALLINT' }, { type: 'INTEGER' }, { type: 'BIGINT' }, () => 0)
```
Comment (gz) [inline web-console/src/lib/compositions/streaming/import/useDefaultRows.ts:14]: e.g., it looks like this is still a `number` type here even though BIGINT is a 64bit value that doesn't fit in a javascript number which is 64-bit floating point

File: web-console/src/lib/functions/common/d3-random-bignumber/exponential.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { BigNumber } from 'bignumber.js'
+
+import { RandomBigNumberGenerationSource } from './generationSource'
+
+/**
```
Comment (gz) [inline web-console/src/lib/functions/common/d3-random-bignumber/exponential.ts:5]: is this copied from soemwhere? if so we should add some attribution to the file and check license

File: web-console/src/lib/functions/common/d3-random-bignumber/generationSource.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export interface RandomBigNumberGenerationSource {
```
Comment (gz) [inline web-console/src/lib/functions/common/d3-random-bignumber/generationSource.ts:1]: same with this file

File: web-console/src/lib/functions/common/d3-random-bignumber/normal.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { BigNumber } from 'bignumber.js'
```
Comment (gz) [inline web-console/src/lib/functions/common/d3-random-bignumber/normal.ts:1]: same with thiese

File: web-console/src/lib/functions/common/valibot/bignumber.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import type { BaseSchema, ErrorMessage, Pipe } from 'valibot'
```
Comment (gz) [inline web-console/src/lib/functions/common/valibot/bignumber.ts:1]: this one too

File: web-console/src/lib/functions/common/valibot/maxBigNumber.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import type { ErrorMessage, PipeResult } from 'valibot'
```
Comment (gz) [inline web-console/src/lib/functions/common/valibot/maxBigNumber.ts:1]: and this one

File: web-console/src/lib/functions/ddl.ts [hunk 9]
```diff
@@ -66,94 +68,76 @@ export function getValueFormatter(columntype: ColumnType): (value: any, params?:
...
+      return new BigNumber(clampBigNumber(min, max, value).toNumber())
     })
-    .with({ type: 'TIME' }, () => {
-      return (value: any) => {
-        // We represent TIME as string for now because the data-grid doesn't
-        // have native support for times yet.
-        // See also
-        return dayjs(value).format('HH:mm:ss')
-      }
+    .with({ type: 'DECIMAL' }, { type: 'NUMERIC' }, () => (value: BigNumber) => {
```
Comment (gz) [inline web-console/src/lib/functions/ddl.ts:107]: NUMERIC can be removed

**Comment by gz** [comment]
The example number that gets generated is no longer showing up on the right side:

Screenshot from 2023-10-27 10-13-42

**Comment by gz** [comment]
If I type a non-number string first and then change it to a valid string the error validation is not reset and it still believes it is invalid:

Screenshot from 2023-10-27 10-15-26

**Comment by gz** [comment]
This is a valid 64bit integer but it is characterized as invalid type:

Screenshot from 2023-10-27 10-16-35

**Comment by gz** [comment]
I'm no longer able to edit the cells of a table directly:
[Screencast from 10-27-2023 10:21:45 AM.webm](https://github.com/feldera/feldera/assets/127654/8c418805-930a-40cd-9322-ab48a360c858)

**Comment by gz** [comment]
I'm unable to delete rows in a table:
[Screencast from 10-27-2023 10:24:23 AM.webm](https://github.com/feldera/feldera/assets/127654/0c6f456d-3291-4340-9f93-e95a0d5faa07)

File: web-console/src/lib/components/streaming/import/RngSettingsDialog.tsx [hunk 10]
```diff
@@ -31,6 +31,7 @@ const RNG_SUPPORTED_TYPES = [
   'DOUBLE',
   'FLOAT',
   'DECIMAL',
+  'NUMERIC',
```
[the remark this answers is not in the document]
Comment (Karakatiza666) [reply web-console/src/lib/components/streaming/import/RngSettingsDialog.tsx:34]: But if a user has a 'NUMERIC' column, they wouldn't see RNG options for it, would they?

[the remark this answers is not in the document]
Comment (mihaibudiu) [reply web-console/src/lib/components/streaming/import/RngSettingsDialog.tsx:34]: NUMERIC and DECIMAL are synonyms

[the remark this answers is not in the document]
Comment (mihaibudiu) [reply web-console/src/lib/components/streaming/import/RngSettingsDialog.tsx:34]: I think the compiler only generates one of them

**Comment by Karakatiza666** [comment]
> I'm no longer able to edit the cells of a table directly by double clicking them: [Screencast from 10-27-2023 10:21:45 AM.webm](https://github.com/feldera/feldera/assets/127654/8c418805-930a-40cd-9322-ab48a360c858)

That's the behavior for the "Browse" tab prior to this PR, too

**Comment by Karakatiza666** [comment]
> Pressing the arrow up / down to increase/decrease the number in the field no longer seems to have an effect:
> 
> Screenshot from 2023-10-27 10-19-02

I had to remove native spin buttons (arrow) from BigNumber input component as I cannot hook into them via JS; I have added ability to change the value with the mouse wheel; LMK if you want me to add custom spin buttons for this input.

**Comment by gz** [comment]
table crash:
[Screencast from 11-03-2023 09:47:44 AM.webm](https://github.com/feldera/feldera/assets/127654/8d55a742-166c-4a3a-8363-6de09d4cdd64)

**Comment by gz** [comment]
> > I'm no longer able to edit the cells of a table directly by double clicking them: [Screencast from 10-27-2023 10:21:45 AM.webm](https://github.com/feldera/feldera/assets/127654/8c418805-930a-40cd-9322-ab48a360c858)
> 
> That's the behavior for the "Browse" tab prior to this PR, too

IMO that's a regression then, since I you made them editable in the past?

**Comment by Karakatiza666** [comment]
> IMO that's a regression then, since I think you made them editable in the past?

I will double check; I thought they were only editable in "Insert" tab

**Comment by gz** [comment]
>I will double check; I thought they were only editable in "Insert" tab

I think you're right I only found a PR for deletion, nevermind then this problem sorry

**Comment by gz** [comment]
adding random row buttons are not working for me:

[Screencast from 11-13-2023 02:23:08 PM.webm](https://github.com/feldera/feldera/assets/127654/7cfd33ae-e84a-4be5-ac74-1c45e8d34496)

**Comment by gz** [comment]
I can't delete the number from the form field, also the example is wrong (should be the constant instead of 0)
[Screencast from 11-13-2023 02:24:06 PM.webm](https://github.com/feldera/feldera/assets/127654/b78cd85d-
6976-4917-88cb-f5e24ddba31f)

**Comment by gz** [comment]
table crash with invariant failed when trying to edit 2nd rng setting field

Screenshot from 2023-11-13 14-25-51

**Comment by gz** [comment]
[Screencast from 11-21-2023 01:18:03 PM.webm](https://github.com/feldera/feldera/assets/127654/56bc657b-140c-4b00-929a-3798eec781e4)

**Comment by gz** [comment]
Aside from the prev bug I posted this lgtm!

**Review by gz — APPROVED** [review]

**Comment by gz** [comment]
I realized there is already an issue for the prev bug: https://github.com/feldera/feldera/issues/1044

**Pull request merged**